fem/bilininteg.cpp

```diff
@@ -907,9 +907,10 @@ void MassIntegrator::AssembleElementMatrix
    for (int i = 0; i < ir->GetNPoints(); i++)
    {
       const IntegrationPoint &ip = ir->IntPoint(i);
-      el.CalcShape(ip, shape);
-
       Trans.SetIntPoint (&ip);
+
+      el.CalcPhysShape(Trans, shape);
+
```

Comment:
Is this change really necessary? Aren't the reference and physical values the same?

Reply:
Not if the `MapType == INTEGRAL`.

Reply:
Ah, true